src/ts/component/page/elements/head/simple.tsx

```diff
@@ -285,6 +299,34 @@ const HeadSimple = observer(class Controls extends React.Component<Props> {
 		return sources.includes(rootId);
 	};
 
+	onCalendar = () => {
+		const { rootId } = this.props;
+		const object = S.Detail.get(rootId, rootId, ['timestamp']);
+
+		S.Menu.open('dataviewCalendar', {
+			element: '#calendar-icon',
+			horizontal: I.MenuDirection.Center,
+			data: {
+				value: object.timestamp,
+				canEdit: true,
+				canClear: false,
+				onChange: (value: number) => {
+					C.ObjectDateByTimestamp(U.Router.getRouteSpaceId(), value, (message: any) => {
+						U.Object.openAuto(message.details);
+					});
+				},
+			},
+		});
+	};
+
+	changeDate = (dir: number) => {
+		const { rootId } = this.props;
+		const object = S.Detail.get(rootId, rootId, ['timestamp']);
```

Comment:
Remove `[ 'timestamp' ]`, add line break. Add timestamp casting to StoreDetail mapDate as Number